File: find-missing-and-repeated-values.ts
{
    function findMissingAndRepeatedValues(grid: number[][]): number[] {
        const set = new Set();

        let dup = 0;
        
        for(const nums of grid) {
            for(const num of nums) {
                if(set.has(num)) {
                    dup = num
                } else {
                    set.add(num)
                }
            }
        }
        
        for(let i = 1; i <= set.size + 1; i++) {
            if(!set.has(i)) {
                return [dup, i]
            }
        }
    };

    // console.log(findMissingAndRepeatedValues([[1,3],[2,2]]))
    console.log(findMissingAndRepeatedValues([[9,1,7],[8,9,2],[3,4,6]]))
}

File: find-missing-and-repeated-values.rs
use std::collections::HashSet;

pub fn find_missing_and_repeated_values(grid: Vec<Vec<i32>>) -> Vec<i32> {
    let mut set: HashSet<i32> = HashSet::new();

    let mut dup = 0;

    for nums in grid {
        for num in nums {
            if set.contains(&num) {
                dup = num;
            } else {
                set.insert(num);
            }
        }
    }

    for i in 1..=set.len() as i32 + 1 {
        if !set.contains(&i) {
            return vec![dup, i];
        }
    }

    return vec![];
}